src/main/java/com/example/bookstore/controller/users/OrdersController.java

```diff
@@ -30,86 +31,99 @@ public class OrdersController {
-	    Integer userId = user.getId(); 
-	    List<CartDetail> cartDetails = cartsService.getCartInfoByUserId(userId);
-	    if (cartDetails.isEmpty()) {
-	    	throw new OrderNotFoundException("Chưa có sản phẩm nào trong giỏ hàng");
-	    }
+	public String newOrder(HttpSession session, Model model) {
+
+		User user = (User) session.getAttribute("user");
+		if (user == null)
+			return "redirect:/sessions/login";
+
+		Integer userId = user.getId();
+		List<CartDetail> cartDetails = cartsService.getCartInfoByUserId(userId);
+		if (cartDetails.isEmpty()) {
+			throw new OrderNotFoundException("Chưa có sản phẩm nào trong giỏ hàng");
+		}
 		double totalPrice = cartDetails.stream().mapToDouble(cd -> cd.getBook().getPrice() * cd.getQuantity()).sum();
-		
+
 		Order order = new Order();
 		order.setUser(user);
-        order.setStatus(Status.PENDING);
-        order.setTotalPrice(totalPrice);
-        
-        for (CartDetail cartDetail : cartDetails) {
-            OrderDetail orderDetail = new OrderDetail();
-            orderDetail.setBook(cartDetail.getBook());
-            orderDetail.setQuantity(cartDetail.getQuantity());
-            orderDetail.setPrice(cartDetail.getBook().getPrice());
-            orderDetail.setOrder(order);
-            order.getOrderDetails().add(orderDetail);
-        }
-        session.setAttribute("order", order);
+		order.setStatus(Status.PENDING);
+		order.setTotalPrice(totalPrice);
+
+		for (CartDetail cartDetail : cartDetails) {
+			OrderDetail orderDetail = new OrderDetail();
+			orderDetail.setBook(cartDetail.getBook());
+			orderDetail.setQuantity(cartDetail.getQuantity());
+			orderDetail.setPrice(cartDetail.getBook().getPrice());
+			orderDetail.setOrder(order);
+			order.getOrderDetails().add(orderDetail);
+		}
+		session.setAttribute("order", order);
 		model.addAttribute("order", order);
-	        
-	    return "users/orders/new";
+
+		return "users/orders/new";
 	}
-	
+
 	@PostMapping("/orders")
 	public String placeOrder(HttpSession session, Model model) {
 
 		User user = (User) session.getAttribute("user");
-	    Order order = (Order) session.getAttribute("order");
-	    if (order == null) {
-	        model.addAttribute("errorMessage", "Đơn hàng không tồn tại. Vui lòng thử lại.");
-	        return "redirect:/order/new";
-	    }
-	    order.setOrderDate(new Date());
-	    
-	    ordersService.saveOrderWithDetails(order, new ArrayList<>(order.getOrderDetails()), user);
-	    
-	    session.removeAttribute("order");
-
-	    return "users/orders/confirmation";
+		Order order = (Order) session.getAttribute("order");
+		if (order == null) {
+			model.addAttribute("errorMessage", "Đơn hàng không tồn tại. Vui lòng thử lại.");
+			return "redirect:/order/new";
+		}
+		order.setOrderDate(new Date());
+
+		ordersService.saveOrderWithDetails(order, new ArrayList<>(order.getOrderDetails()), user);
+
+		session.removeAttribute("order");
+
+		return "users/orders/confirmation";
+	}
+
+	@GetMapping("/details/{orderId}")
```

Comment:
```suggestion
	@GetMapping(/{orderId}")
```
vẽ thêm chi cho bị comment bé bì